### ISSUE TITLE
Some commands trash `context.enable` setting

### ISSUE THREAD

**Issue by Grazfather** [hugsy/gef#299]
Some commands trash context.enable. e.g. `pie run`

```
b <somefunc>
gef config context.enable False
pie run
```

**Comment by Grazfather** [comment]
This is obviously because they call `context_enable`. I think the easiest fix is just to have the context display based on TWO values: the 'gef' version of enabled (e.g. for reasons) and the user version (based on the setting). `context_enable` should set the former flag, and not change a user's settings.

**Comment by hugsy** [comment]
Why not simply get the current value of `context.enable` to decide if we should call `enable_context()` ?

**Comment by hugsy** [comment]
Yeah, pretty sure a patch likethis should do it:

```python
        should_display_context = get_gef_setting("context.enable")
        [...]
        disable_context()
        gdb.execute("run {}".format(" ".join(argv)))
        if should_display_context:
            enable_context()
```

**Comment by hugsy** [comment]
Hum../ Not a big fan of having 2 pieces of code doing the same thing, it's confusing and prone to mistake (especially by non regular contributors).

Feel free to PoC it, I can have a look.

**Comment by Grazfather** [comment]
What do you mean two pieces of code? there would no longer be a need for `enable_context`, since that would ONLY be a setting now.

**Comment by Grazfather** [comment]
Care to check out 70ccaa0?

**Comment by Grazfather** [comment]
Fixed with 70ccaa0
